mitosheet/src/components/endo/celleditor/cellEditorUtils.tsx

```diff
@@ -319,4 +349,59 @@ export const getDocumentationFunction = (formula: string): FunctionDocumentation
     } else {
         return matchingFunctions[0];
     }
+}
+
+export const getNewIndexLabelAtRowOffsetFromOtherIndexLabel = (sheetData: SheetData, indexLabel: IndexLabel | undefined, rowOffset: number): IndexLabel | undefined => {
+    if (indexLabel === undefined) {
+        return undefined;
+    }
+    
+    const indexOfIndexLabel = sheetData.index.indexOf(indexLabel);
+    if (indexOfIndexLabel === -1) {
+        return undefined;
+    }
+
+    const indexOfNewLabel = indexOfIndexLabel - rowOffset;
+    return sheetData.index[indexOfNewLabel];
+}
+
+
+export const getFormulaStringFromFrontendFormula = (formula: Formula | undefined, indexLabel: IndexLabel | undefined, sheetData: SheetData | undefined): string | undefined => {
+    let formulaString = '';
+    if (!formula || !sheetData) {
+        return formulaString;
+    }
+
+    formula.forEach(formulaPart => {
+        if (formulaPart.type === 'string part') {
+            formulaString += formulaPart.string
+        } else {
+            formulaString += formulaPart.display_column_header
+
+            /**
+             * After adding the reference to the column header, we need to add the correct index label.
+             * Notably, this is the index label that is the formulaPart.rowOffset from the current indexLabel
+             */
+            const newIndexLabel = getNewIndexLabelAtRowOffsetFromOtherIndexLabel(sheetData, indexLabel, formulaPart.row_offset);
+            if (newIndexLabel !== undefined) {
+                formulaString += getDisplayColumnHeader(newIndexLabel);
+            } else {
+                /**
+                 * TODO: how do we want to handle the case where the newIndexLabel is undefined? This happens when 
+                 * we might have written a formula that references the row before, but now we're editing in the first row. 
+                 * 
+                 * E.g. B1 = A1 + A0, then go to B0. What do you see? B0 = A0 + A(?)
+                 * 
+                 * Currently, we handle this just by clamping to the first index, but this can be somewhat misleading. Let
+                 * me know what you think we should do here... we could try putting in the fill_value for the shift function...
```

Comment:
Excel makes the formula `=#REF!`

I think you're right that its a bit misleading as is because if a user goes to figure our why its `NaN` which would be super reasonable thing to do, they'll end up changing the formula. 

In this example: eg. B1 = A1 + A0, then go to B0. What do you see? B0 = A0 + A(?)

Some ideas: 
= A0 + A-1
= A0 + #INVALID_REFERENCE
= #INVALID_REFERENCE
= #REF!

I would be fine with any of these.

Reply:
I'm going to just make it so it puts a 0 in there. For most use cases, I think this is a reasonable thing to display and do - as it actually gives the users the input that is in there!

Reply:
This notably has the tradeoff that if the user goes and tries to submit this formula, it will change the formula in the cell. But I think this is a suitable tradeoff - as it has the benefit of actually being more clear when the user is looking at it!